ethstorage/p2p/node.go

```diff
@@ -193,7 +194,8 @@ func (n *NodeP2P) RequestL2Range(ctx context.Context, start, end uint64) (uint64
 // RequestShardList fetches shard list from remote peer
 func (n *NodeP2P) RequestShardList(remotePeer peer.ID) ([]*protocol.ContractShards, error) {
 	remoteShardList := make([]*protocol.ContractShards, 0)
-	s, err := n.Host().NewStream(context.Background(), remotePeer, protocol.RequestShardList)
+	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Do we want to add the timeout for all the others request?

Reply:
better handle cancel? like
```
ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
defer cancel()
```